app/src/main/java/com/example/myapplication/DepthTransformation.java

```diff
@@ -0,0 +1,36 @@
+package com.example.myapplication;
+
+import android.view.View;
+
+import androidx.viewpager.widget.ViewPager;
+
+public class DepthTransformation implements ViewPager.PageTransformer{
+    @Override
+    public void transformPage(View page, float position) {
```

Comment:
I would document this code segment, to understand the purpose of this class.
Maybe extract the code in each block to different method with an informative name.
Maybe some constants instead of magic numbers, for example POS_CENTER = 0;